### ISSUE TITLE
ObjectMapper.convertValue() should throw JsonProcessingException

### ISSUE THREAD

**Issue by ewirch** [FasterXML/jackson-databind#3586]
Version: 2.13.1

Doc of `treeToValue()` says:

> Functionally equivalent to: objectMapper.convertValue(n, valueClass);
> Note: inclusion of {@code throws JsonProcessingException} is not accidental since while there can be no input decoding problems, it is possible that content does not match target type

The same applies to `convertValue()`. I think `convertValue` catches `JsonProcessingException` by accident:
```
} catch (IOException e) { // should not occur, no real i/o...
      throw new IllegalArgumentException(e.getMessage(), e);
}
```
`JsonProcessingException` is not an i/o problem, but still a sub class of `IOException`.

**Comment by cowtowncoder** [comment]
Thank you for reporting this. Yes, ideally`JsonProcessingException` should be passed through by `convertValue()` but unfortunately existing method signature does not expose either `IOException` or `JsonProcessingException`.
And due to backwards-compatibility requirements (adding new exceptions in signature is source-incompatible change although not binary-incompatible, I think) this cannot really be changed until Jackson 3.0.
Jackson 3.0 will change `JsonProcessingException` to be unchecked exception so this issue will disappear.

So, things work the way they have to here.

**Comment by cowtowncoder** [comment]
Marking is resolved for 3.0; cannot be changed for 2.x.
